packages/config/src/validate/validations.js

```diff
@@ -91,6 +92,17 @@ const POST_NORMALIZE_VALIDATIONS = [
     message: 'must be a string.',
     example: () => ({ plugins: [{ package: 'netlify-plugin-one' }] }),
   },
+  // We don't allow `package@tag|version` nor `git:...`, `github:...`,
+  // `https://...`, etc.
+  // We skip this validation for local plugins.
+  // We ensure @scope/plugin still work.
+  {
+    property: 'plugins.*.package',
+    check: package =>
+      package.startsWith('.') || package.startsWith('/') || validateNpmPackageName(package).validForOldPackages,
```

Comment:
Note: `validForOldPackages` is `false` when using `package@version` or `git:...`